aravis/src/manual/register.rs

```diff
@@ -0,0 +1,44 @@
+use ffi::arv_gc_register_get;
+use glib::object::IsA;
+use glib::translate::*;
+use std::ffi::c_void;
+use std::ptr;
+
+use crate::GcRegister;
+
+pub(crate) mod traits {
+	pub trait GcRegisterExtManual {
+		fn get(&self, buffer: &mut [u8]) -> Result<(), glib::Error>;
+		fn set(&self, buffer: &[u8]) -> Result<(), glib::Error>;
+	}
+}
+
+impl<T: IsA<GcRegister>> traits::GcRegisterExtManual for T {
+	fn get(&self, buffer: &mut [u8]) -> Result<(), glib::Error> {
+		unsafe {
+			let mut error = ptr::null_mut();
+			let len = buffer.len();
+			let ptr = buffer.as_mut_ptr();
+			arv_gc_register_get(self.as_ref().to_glib_none().0, ptr as *mut c_void, len as u64, &mut error);
+			if error.is_null() {
+				Ok(())
+			} else {
+				Err(from_glib_full(error))
+			}
+		}
+	}
+
+	fn set(&self, buffer: &[u8]) -> Result<(), glib::Error> {
+		unsafe {
+			let mut error = ptr::null_mut();
+			let len = buffer.len();
+			let ptr = buffer.as_ptr();
+			ffi::arv_gc_register_set(self.as_ref().to_glib_none().0, ptr as *mut c_void, len as u64, &mut error);
```

Comment:
This feels wrong to me, and I am sort of inclined to require a mutable slice rather than an immutable one since we need to use `*mut c_void`.

Reply:
Well, the reality is that in C people often forget about `const`, so we shouldn't worry about it too much. You could open an issue or PR at the main aravis project to fix the signature if you really want it fixed :)

Reply:
> Well, the reality is that in C people often forget about const

Yes they do. But what is wrong about the declaration of arv_gc_register_set ? The buffer is declared as const.

Reply:
Apologies, I just assumed a missing `const` was the most probable explanation. But apparantly it's the `gir` tool from `rust-gtk` that is making a mistake. Or perhaps there is a problem with the generated introspection file.

Also, thanks again for the Aravis project! I'm still very happy to be able to avoid the closed source overly-complex blobs from the EMVA :D 

@dan-jfisher: if you feel like figuring out why our bindings are `mut *` instead of `const *`, knock yourself out. Otherwise, just ignore it :)

Reply:
I'll look into it again when I have some downtime, for now I will stick with blind faith.